spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -192,6 +194,14 @@ public Map<String, String> extraSnapshotMetadata() {
           }
         });
 
+    final Iterator<Tuple2<String, String>> it = sessionConf.getAll().iterator();
+    while (it.hasNext()) {
+      final Tuple2<String, String> entry = it.next();
+      if (entry._1.startsWith("spark." + SnapshotSummary.EXTRA_METADATA_PREFIX)) {
+        extraSnapshotMetadata.put(
+            entry._1.substring(6 + SnapshotSummary.EXTRA_METADATA_PREFIX.length()), entry._2);
```

Comment:
Can you make private static constants rather than embedding `6` and `spark.`?

Reply:
done